mempool/src/shared_mempool.rs

```diff
@@ -173,14 +236,51 @@ async fn sync_with_peers<'a>(
     }
 }
 
-/// used to validate incoming transactions and add them to local Mempool
+fn convert_txn_from_proto(txn_proto: SignedTransactionProto) -> Option<SignedTransaction> {
+    match SignedTransaction::try_from(txn_proto.clone()) {
+        Ok(txn) => Some(txn),
+        Err(e) => {
+            security_log(SecurityEvent::InvalidTransactionMP)
+                .error(&e)
+                .data(&txn_proto)
+                .log();
+            None
+        }
+    }
+}
+
+/// submits a list of SignedTransaction to the local mempool
+/// and returns a vector containing AdmissionControlStatus
 async fn process_incoming_transactions<V>(
     smp: SharedMempool<V>,
-    peer_id: PeerId,
+    // ID of the peer that submitted the transactions. None if transaction was submitted by AC client
+    peer_id: Option<PeerId>,
```

Comment:
can we instead just pass timeline_state here?
e.g. for client events it's `TimelineState::NotReady`
still not sure if I like Option<PeerId>